### PR TITLE
fix #84 : 0-size arrays like LArray([]) are not displayed.

### LINKED ISSUE #84: fix New

### PR DESCRIPTION
**Pull request by alixdamman**

### CODE REVIEW

File: larray_editor/arraywidget.py [hunk 1]
```diff
@@ -506,9 +506,9 @@ def __init__(self, parent, data_scrollbar):
 
 
 class ArrayEditorWidget(QWidget):
-    def __init__(self, parent, data, readonly=False, bg_value=None, bg_gradient=None, minvalue=None, maxvalue=None):
+    def __init__(self, parent, data=None, readonly=False, bg_value=None, bg_gradient=None, minvalue=None, maxvalue=None):
         QWidget.__init__(self, parent)
-        readonly = np.isscalar(data)
+        readonly = data is not None and np.isscalar(data)
```
Comment (gdementen) [inline larray_editor/arraywidget.py]: This seems odd. why do you need readonly=False when data is None?

File: larray_editor/arrayadapter.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def get_axes_names(self):
 
     def get_axes(self):
         axes = self.filtered_data.axes
-        if len(axes) == 0:
+        if self.filtered_data.size == 0 or len(axes) == 0:
```
Comment (gdementen) [inline larray_editor/arrayadapter.py]: `or len(axes) == 0` seems superfluous. I can see how you can have `size == 0` with `len(axes) > 0`, but I don't see any case where you can have `size > 0` with `len(axes) == 0`. You could also write: `if not self.filtered_data.size:` but that is a matter of taste.

File: larray_editor/arraywidget.py (hunk 1 above)
Comment (alixdamman) [reply larray_editor/arraywidget.py]: This line is historical. I thought it was there for a good reason so I let it.

File: larray_editor/arrayadapter.py [hunk 3]
```diff
@@ -35,29 +35,28 @@ def get_axes_names(self):
...
             return [[axis_name] for axis_name in axes_names]
 
     def get_xlabels(self):
-        axes = self.filtered_data.axes
-        if len(axes) == 0:
+        if self.filtered_data.size == 0:
             return None
-        elif len(axes.labels[-1]) == 0:
+        axes = self.filtered_data.axes
+        if len(axes.labels[-1]) == 0:
```
Comment (gdementen) [inline larray_editor/arrayadapter.py]: I wonder if this case is possible. In other words, is it possible to have len(axes.labels[-1]) == 0 with self.filtered_data.size > 0 ?

**Pull request merged**